spec/models/patient_spec.rb

```diff
@@ -693,5 +693,22 @@ def refresh_views
 
       expect(Teleconsultation.where(patient: patient)).to be_empty
     end
+
+    it "soft deletes the patient's notifications" do
+      notification = create(:notification)
+      notification.patient.discard_data
+      expect(Notification.where(id: notification.id)).to be_empty
+    end
+
+    it "soft deletes the patient's membership in any experiment" do
+      patient = create(:patient)
+      experiment = create(:experiment, :with_treatment_group)
+      treatment_group = experiment.treatment_groups.first
+      treatment_group.patients << patient
+
+      patient.discard_data
+      # TODO: Fails because TreatmentGroupMembership doesn't have a 'deleted_at' column, and is not an ApplicationRecord
```

Comment:
Oof, this is a doozy. Are you thinking we destroy the record instead, or add the necessary timestamp?